lib/libc/minimal/source/stdout/prf.c

```diff
@@ -663,35 +649,29 @@ int z_prf(int (*func)(), void *dest, const char *format, va_list vargs)
 					*va_arg(vargs, int *) = count;
 					break;
 				}
-				break;
+				continue;
 
 			case 'p':
 				val = (uintptr_t) va_arg(vargs, void *);
 				c = _to_hex(buf, val, true, 2*sizeof(void *), 'x');
-				need_justifying = true;
-				if (precision != -1) {
+				if (precision >= 0) {
 					pad = ' ';
 				}
 				break;
 
 			case 's':
-			{
-				char *cptr_temp = va_arg(vargs, char *);
+				cptr = va_arg(vargs, char *);
 				/* Get the string length */
-				for (c = 0; c < MAXFLD; c++) {
-					if (cptr_temp[c] == '\0') {
+				if (precision < 0) {
+					precision = __INT_MAX__;
```

Comment:
if you are using `__INT_MAX__` would be nice to explicitly include `limits.h` in this file.